### ISSUE TITLE
[CCKP Toolkit] New Tool Schema - Master ticket

### ISSUE THREAD

**Issue by aditigopalan** [mc2-center/data-models#120]
Using this as the master ticket to track the steps following the integration of the CCKP toolkit into the data model:

- [x] Release the new tool schema. #121 
- [ ] Plan backpopulation efforts due to changes in the tool schema (some new fields are now required).
- [ ] Review changes to union tables and the union table workflow.
- [ ] Assess updates to syncing scripts for the portal.
- [ ] Plan and modify the tool display for the CCKP as needed.

**Comment by aclayton555** [comment]
24-9: Will be presenting on both Sept 23 SC meeting (this is an opportunity for feedback) AND during Oct 4 site visit (opportunity to report back on progress, direction and feedback received so far). One things to share is the idea for a guideline that was brought up by the data managers. This will deliver value through a proactive approach to help the community submit tools (rather than check than fix).

**Comment by aclayton555** [comment]
24-9 close-out: @aditigopalan will update description and scoping of this ticket following engagement with DPE. Expecting work to continue in next sprint.

**Comment by aclayton555** [comment]
24-10 close-out: Some parallel engineering work needed here and additional input with Gregory Way. Additional thought needed into how to potentially also surface the testing output on the details page. Data modelling aspects on hold for now, but next effort will focus on RFC in pilot with #97

**Comment by aclayton555** [comment]
24-11/12 Ongoing development discussions with Way team. This work will continue into 2025

**Comment by aditigopalan** [comment]
25-1 sprint: Ongoing development, integrated with external tool, working on deploying on tower (setting this up with Tom/ Brad), will work with portals team in Q2 to get this up

**Comment by aditigopalan** [comment]
[CCKP mockup](https://docs.google.com/presentation/d/1hr01RxbPENK4BAgOgoPCP2E63cPvI2bPj10g9a-2oW8/edit#slide=id.g32a0a873318_0_29)
[Synapse folder](https://www.synapse.org/Synapse:syn64626421) where results are stored

**Comment by aclayton555** [comment]
25-1 Close-out:
- Next step to engage with Design and understand if we can get mockups by AACR; expected portal work would not occur until Q2 (see https://sagebionetworks.jira.com/browse/CPO-1458)
- Need to determine where we would store these metadata (currently a JSON). Could be converted to a Synapse table and pushed this to the grant-specific projects/toolview tables OR we could have join of the existing Toolview tables and these data (creating an additional table to power these data / link tables based on name). Leaning toward the latter approach so that contributors cannot edit these data. Overall, flexibility in the approach but the design and implementation consult may introduce constraints, so approach design first
- Note: Toolkit pulls the URL for the tool (e.g. GH url) not the Synapse ID
- In parallel with our engagement with Design, Aditi can continue running tools through workflow to collect data.

**Comment by Bankso** [comment]
Wanted to note some updates related to this ticket:
- updated tool schema has been incorporated into the [MC2 Data Model](https://github.com/mc2-center/data-models/blob/main/modules/tool/annotationProperty.csv)
- existing tool metadata table schemas have been updated and changes have propagated to [ToolView union table](https://www.synapse.org/Synapse:syn52820451/tables/)
- [CCKP Tools Merged table schema](https://www.synapse.org/Synapse:syn26127427/tables/) has been updated to reflect current model

**Comment by aditigopalan** [comment]
Tracking now on [toolkit Github](https://github.com/mc2-center/cckp-toolkit-workflow) and Jira
